connectors/src/connectors/slack/bot.ts

```diff
@@ -0,0 +1,188 @@
+    logger.error(
+      {
+        error: res.error,
+        slackTeamId: slackTeamId,
+        slackChannel: slackChannel,
+        slackUserId: slackUserId,
+        slackMessageTs: slackMessageTs,
+      },
+      "Failed answering to Slack Chat Bot message"
+    );
+    const accessToken = await getAccessToken(connector.connectionId);
+    const slackClient = getSlackClient(accessToken);
+    await slackClient.chat.postMessage({
+      channel: slackChannel,
+      text: `An error occured. Our team has been notified and will work on it as soon as possible.`,
+      thread_ts: slackMessageTs,
+    });
+  }
+
+  return res;
+}
+
+async function botAnswerMessage(
+  message: string,
+  slackTeamId: string,
+  slackChannel: string,
+  slackUserId: string,
+  slackMessageTs: string,
+  connector: Connector
+): Promise<Result<ChatSessionUpdateEvent, Error>> {
+  const { DUST_API = "https://dust.tt" } = process.env;
+
+  const slackChatBotMessage = await SlackChatBotMessage.create({
+    connectorId: connector.id,
+    message: message,
+    slackUserId: slackUserId,
+    slackEmail: "",
+    slackUserName: "",
+    channelId: slackChannel,
+    messageTs: slackMessageTs,
+  });
+
+  const accessToken = await getAccessToken(connector.connectionId);
+  const slackClient = getSlackClient(accessToken);
+  const slackUserInfo = await slackClient.users.info({
+    user: slackUserId,
+  });
+
+  if (!slackUserInfo.ok || !slackUserInfo.user) {
+    throw new Error(`Failed to get user info: ${slackUserInfo.error}`);
+  }
+  const slackUser = slackUserInfo.user;
+  if (slackUser.profile?.email) {
+    slackChatBotMessage.slackEmail = slackUser.profile.email;
+  }
+  if (slackUser.profile?.display_name) {
+    slackChatBotMessage.slackUserName = slackUser.profile.display_name;
+  }
+  await slackChatBotMessage.save();
+  const c = new DustAPI({
+    workspaceId: connector.workspaceId,
+    apiKey: connector.workspaceAPIKey,
+  });
+  const mainMessage = await slackClient.chat.postMessage({
+    channel: slackChannel,
+    text: "_I am thinking..._",
+    thread_ts: slackMessageTs,
+    mrkdwn: true,
+  });
+
+  const matches = message.match(/<@[A-Z-0-9]+>/g);
+  if (matches) {
+    const mySlackUser = await whoAmI(accessToken);
+    for (const m of matches) {
+      const userId = m.replace(/<|@|>/g, "");
+      if (userId === mySlackUser) {
+        message = message.replace(m, "");
+      } else {
+        const userName = await getUserName(
+          userId,
+          connector.id.toString(),
+          slackClient
+        );
+        message = message.replace(m, `@${userName}`);
+      }
+    }
+  }
+  const chatRes = await c.newChatStreamed(
+    message,
+    slackUser.tz ? slackUser.tz : "Europe/Paris"
+  );
+  if (chatRes.isErr()) {
+    return new Err(new Error(chatRes.error.message));
+  }
+  const chat = chatRes.value;
+  let fullAnswer = "";
+  let lastSentDate = new Date();
+  for await (const event of chat.eventStream) {
```

Comment:
You're probably missing the `chat_message_create` event in case it is of type `error` otherwise you'll miss most errors. In this case there is no tokens stream and a `chat_message_create` is created with role `error`. No `chat_session_create` is sent.